plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -200,44 +203,22 @@ public void setupLaunchConfiguration(ILaunchConfigurationWorkingCopy workingCopy
     workingCopy.setMappedResources(projects.toArray(new IResource[projects.size()]));
   }
 
-  private void checkAndSetPorts() throws CoreException {
-    checkAndSetPorts(getServer(), new PortProber() {
-      @Override
-      public boolean isPortInUse(int port) {
-        return SocketUtil.isPortInUse(port);
-      }
-    });
-  }
-
-  @VisibleForTesting
-  public interface PortProber {
-    boolean isPortInUse(int port);
-  }
-
+  /**
+   * @param portProber returns true if the argument is a port that is in use
+   * @return the port value ⊂ [0, 65535]
+   */
   @VisibleForTesting
-  void checkAndSetPorts(IServer server, PortProber portProber) throws CoreException {
-    serverPort = checkPortAttribute(server, portProber,
-        SERVER_PORT_ATTRIBUTE_NAME, DEFAULT_SERVER_PORT);
-    adminPort = checkPortAttribute(server, portProber,
-        ADMIN_PORT_ATTRIBUTE_NAME, DEFAULT_ADMIN_PORT);
-  }
-
-  private static int checkPortAttribute(IServer server, PortProber portProber,
-      String attribute, int defaultPort) throws CoreException {
-    int port = server.getAttribute(attribute, defaultPort);
+  static int checkPort(Integer portOrNull, int defaultValue,
+      Predicate<Integer> portProber)
```

Comment:
I find this a bit confusing. How about `portProber` --> `portInUse`? (Javadoc too.)